File: app/src/main/java/me/imzack/app/end/injector/component/ReminderComponent.java
package me.imzack.app.end.injector.component;

import me.imzack.app.end.injector.module.ReminderPresenterModule;
import me.imzack.app.end.injector.scope.ActivityScope;
import me.imzack.app.end.view.activity.ReminderActivity;

import dagger.Component;

@ActivityScope
@Component(dependencies = AppComponent.class, modules = ReminderPresenterModule.class)
public interface ReminderComponent {

    void inject(ReminderActivity reminderActivity);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/ReminderComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.ReminderPresenterModule
import net.zackzhang.app.end.injector.scope.ActivityScope
import net.zackzhang.app.end.view.activity.ReminderActivity

@ActivityScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(ReminderPresenterModule::class))
interface ReminderComponent {

    fun inject(reminderActivity: ReminderActivity)
}
